internal/handlers/pg/pgdb/query_documents.go

```diff
@@ -126,7 +133,7 @@ func (pgPool *Pool) QueryDocuments(ctx context.Context, querier pgxtype.Querier,
 
 // iterateFetch iterates over the rows returned by the query and sends FetchedDocs to fetched channel.
 // It returns ctx.Err() if context cancellation was received.
-func iterateFetch(ctx context.Context, fetched chan FetchedDocs, rows pgx.Rows) error {
+func iterateFetch(ctx context.Context, fetched chan FetchedDocs, rows pgx.Rows, explain bool) error {
```

Comment:
Hmm, I wonder if it makes sense to have a totally separate function for `explain`. If I understand your code (and how Postgres works) correctly, we won't have the results, we will only have the plan. (It's not `explain analyze`, it's only `explain`). So then the whole story with iterating through lots of docs doesn't make a lot of sense. Also, there is not a lot of common code between the case with `explain` and without it.

Reply:
> Hmm, I wonder if it makes sense to have a totally separate function for `explain`. If I understand your code (and how Postgres works) correctly, we won't have the results, we will only have the plan. (It's not `explain analyze`, it's only `explain`). 

The plan is the result. We have results. WDYM? I don't understand.

> So then the whole story with iterating through lots of docs doesn't make a lot of sense. 

It's not a lot of docs, right, but when we would build an explain for queries where pushdown happens (not only for `_id` but for other complex requests, it will be more convenient to have it right where the SQL query creation happens. 

> Also, there is not a lot of common code between the case with `explain` and without it.

That's for now, please see the previous message.

Reply:
I'll try to extract a func that builds query then probably it's possible to do it in a separate function.

Reply:
@rumyantseva please see the result if it satisfies you.